plugin/trino-influxdb/src/test/java/io/trino/plugin/influxdb/server/InfluxServer.java

```diff
@@ -0,0 +1,130 @@
+    public static final String USERNAME = "testuser";
+    public static final String PASSWORD = "testpass";
+    private static final Logger log = Logger.get(InfluxServer.class);
+    private static final int PORT = 8086;
+    private GenericContainer<?> dockerContainer;
+    private InfluxSession session;
+
+    public InfluxServer()
+            throws Exception
+    {
+        this(VERSION);
+    }
+
+    public InfluxServer(String influxVersion)
+            throws Exception
+    {
+        log.info("Influx server starting...");
+
+        DockerImageName dockerImageName = DockerImageName.parse("influxdb").withTag(influxVersion);
+        try {
+            this.dockerContainer = new InfluxDBContainer<>(dockerImageName)
+                    .withExposedPorts(PORT)
+                    .withAdmin(ADMIN_USERNAME)
+                    .withAdminPassword(ADMIN_PASSWORD)
+                    .withUsername(USERNAME)
+                    .withPassword(PASSWORD);
+            this.dockerContainer.start();
+
+            InfluxSession session = new InfluxSession(getHost(), getPort());
+            this.checkConnectivity(session);
+            this.initializeData(session);
+            this.session = session;
+        }
+        catch (Exception e) {
+            log.error("Influx server start error. %s", e);
+            if (this.session != null) {
+                this.session.close();
+            }
+            if (this.dockerContainer != null) {
+                this.dockerContainer.close();
+            }
+            throw e;
+        }
+
+        log.info("Influx server started.");
+    }
+
+    public String getHost()
+    {
+        return this.dockerContainer.getHost();
+    }
+
+    public int getPort()
+    {
+        return this.dockerContainer.getMappedPort(PORT);
+    }
+
+    public Map<String, String> buildConnectorProperties()
+    {
+        Map<String, String> connectorProperties = new HashMap<>();
+        connectorProperties.put("influx.host", this.getHost());
+        connectorProperties.put("influx.port", Integer.toString(this.getPort()));
+        connectorProperties.put("influx.protocol", PROTOCOL);
+        connectorProperties.put("influx.username", USERNAME);
+        connectorProperties.put("influx.password", PASSWORD);
+        connectorProperties.put("influx.hasTimeField", "false");
+        return connectorProperties;
+    }
+
+    @Override
+    public void close()
+    {
+        if (this.session != null) {
+            this.session.close();
+        }
+        this.dockerContainer.close();
+    }
+
+    @ResourcePresence
+    public boolean isRunning()
+    {
+        return dockerContainer.getContainerId() != null;
+    }
+
+    private void checkConnectivity(InfluxSession session)
+    {
+        String version = session.checkConnectivity();
+        log.info("Influx server connectivity check is %s.", !isNullOrEmpty(version) ? "success" : "fail");
+    }
+
+    private void initializeData(InfluxSession session)
```

Comment:
This method always create timeseries data, but it's redundant when such data isn't required in tests. Please avoid adding such logic to testing server class.